cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -185,24 +185,55 @@ def _parameter_names_(self) -> AbstractSet[str]:
             )
         }
 
-    def _decompose_(self) -> 'cirq.OP_TREE':
-        result = self.circuit.unfreeze()
-        result = result.transform_qubits(lambda q: self.qubit_map.get(q, q))
+    def mapped_circuit(self, deep: bool = False) -> 'cirq.Circuit':
+        circuit = self.circuit.unfreeze()
+        circuit = circuit.transform_qubits(lambda q: self.qubit_map.get(q, q))
         if self.repetitions < 0:
-            result = result ** -1
-        result = protocols.with_measurement_key_mapping(result, self.measurement_key_map)
-        result = protocols.resolve_parameters(result, self.param_resolver, recursive=False)
-        # repetition_ids don't need to be taken into account if the circuit has no measurements
-        # or if repetition_ids are unset.
-        if self.repetition_ids is None or not protocols.is_measurement(result):
-            return list(result.all_operations()) * abs(self.repetitions)
-        # If it's a measurement circuit with repetitions/repetition_ids, prefix the repetition_ids
-        # to measurements. Details at https://tinyurl.com/measurement-repeated-circuitop.
-        ops = []  # type: List[cirq.Operation]
-        for repetition_id in self.repetition_ids:
-            path = self.parent_path + (repetition_id,)
-            ops += protocols.with_key_path(result, path).all_operations()
-        return ops
+            circuit = circuit ** -1
+        has_measurements = protocols.is_measurement(circuit)
+        if has_measurements:
+            circuit = protocols.with_measurement_key_mapping(circuit, self.measurement_key_map)
+        circuit = protocols.resolve_parameters(circuit, self.param_resolver, recursive=False)
+        if deep:
+
+            def remap_op_fn(op: 'cirq.Operation'):
```

Comment:
nit: add return type
```suggestion
            def remap_op_fn(op: 'cirq.Operation') -> 'cirq.Operation':
```
Also, not sure what the "re" means here; perhaps call this `map_deep` instead?

Reply:
Done. Return type aligns with the type of the `func` arg in `circuit.map_ops`.